Spec/RealtimeClientPresence.swift

```diff
@@ -3590,6 +3590,174 @@ class RealtimeClientPresence: QuickSpec {
+
+                        waitUntil(timeout: testTimeout) { done in
+                            transport.afterProcessingReceivedMessage = { protocolMessage in
+                                // Receive the first Sync message from Ably service
+                                if protocolMessage.action == .sync {
+
+                                    // Inject a fabricated Presence message
+                                    let presenceMessage = ARTProtocolMessage()
+                                    presenceMessage.action = .presence
+                                    presenceMessage.channel = protocolMessage.channel
+                                    presenceMessage.connectionSerial = protocolMessage.connectionSerial + 1
+                                    presenceMessage.timestamp = NSDate() as Date
+                                    presenceMessage.presence = presenceData
+
+                                    transport.receive(presenceMessage)
+
+                                    // Simulate an end to the sync
+                                    let endSyncMessage = ARTProtocolMessage()
+                                    endSyncMessage.action = .sync
+                                    endSyncMessage.channel = protocolMessage.channel
+                                    endSyncMessage.channelSerial = "validserialprefix:" //with no part after the `:` this indicates the end to the SYNC
+                                    endSyncMessage.connectionSerial = protocolMessage.connectionSerial + 2
+                                    endSyncMessage.timestamp = NSDate() as Date
+
+                                    transport.afterProcessingReceivedMessage = nil
+                                    transport.receive(endSyncMessage)
+
+                                    // Stop the next sync message from Ably service because we already injected the end of the sync
+                                    transport.actionsIgnored = [.sync]
+
+                                    done()
+                                }
+                            }
+                            channelSubscribed.attach()
+                        }
+
+                        let query = ARTRealtimePresenceQuery()
+                        query.connectionId = "one"
+
+                        waitUntil(timeout: testTimeout) { done in
+                            channelSubscribed.presence.get(query) { members, error in
+                                expect(error).to(beNil())
+                                guard let members = members else {
+                                    fail("Members is nil"); done(); return
+                                }
+                                expect(members).to(haveCount(2))
+                                expect(members).to(allPass({ (member: ARTPresenceMessage?) in member!.action != .absent }))
+                                expect(members.filter{ $0.clientId == "a" }).to(beEmpty())
```

Comment:
How will this be empty, given that, in `presenceData`, messages with client ID `a` also have connection ID `one`?

Reply:
@tcard client ID `a` has a `leave` action.

Sorry, I copy&pasted the mock data from another test and maybe that's confusing you.

Reply:
Ah, I see the timestamps. It may be a bit clearer if the order in the array was the chronological order, but never mind.